src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackageCommandRunner.cs

```diff
@@ -0,0 +1,362 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Evaluation;
+using NuGet.CommandLine.XPlat.Utility;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+/// <summary>
+/// A struct to simplify holding all of the information
+/// about a package reference when using list
+/// </summary>
+public struct PRPackage
+{
+    public string package;
+    public VersionRange requestedVer;
+    public string requestedVerStr;
+    public NuGetVersion resolvedVer;
+    public NuGetVersion suggestedVer;
+    public bool deprecated;
+    public bool autoRef;
+}
+
+namespace NuGet.CommandLine.XPlat
+{
+    public class ListPackageCommandRunner : IListPackageCommandRunner
+    {
+        private bool _autoReferenceFound = false;
+
+        public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs, MSBuildAPIUtility msBuild)
+        {
+
+            //If the given file is a solution, get the list of projects
+            //If not, then it's a project, which is put in a list
+            var projectsPaths = Path.GetExtension(listPackageArgs.Path).Equals(".sln")
+                           ?
+                           MSBuildAPIUtility.GetProjectsFromSolution(listPackageArgs.Path)
+                           .Where(f => File.Exists(f))
+                           :
+                           new List<string>(new string[] { listPackageArgs.Path });
+
+            //Loop through all of the project paths
+            foreach (var projectPath in projectsPaths)
+            {
+                //Open project to evaluate properties
+                var project = MSBuildAPIUtility.GetProject(projectPath);
+                var projectName = project.GetPropertyValue("MSBuildProjectName");
+
+                //Get all the packages that are referenced in a project
+                var packages = msBuild.GetPackageReferencesForList(project, projectPath, listPackageArgs.Frameworks, listPackageArgs.Transitive);
```

Comment:
You just need to pass `projectPath`, `framework` and `isTransitive` to `GetPackageReferencesForList` api and you should be good.

Reply:
I use the project to get the assets file and get the requested versions for the packages. If I didn't pass `project` I'll need to open the project again to get the utility file, which seems unnecessary.
I actually can do without `projectPath` if I pass `project`